### PR TITLE
Refactor widgets and fix lint issues

### PR DESCRIPTION
**Pull request by pixeldesu**
# What

This PR fixes most of the lint issues in `packages/client/src/widgets`

# Why

Some elements were not in the Composition API, so that has been adjusted, aside that, mostly events and error handlers have been fixed to not use `e` as a name.

### CODE REVIEW

File: packages/client/src/widgets/activity.calendar.vue [hunk 1]
```diff
@@ -23,45 +23,41 @@
...
-	props: ['data'],
-	created() {
-		for (const d of this.data) {
-			d.total = d.notes + d.replies + d.renotes;
-		}
-		const peak = Math.max.apply(null, this.data.map(d => d.total));
+for (const d of props.activity) {
+	d.total = d.notes + d.replies + d.renotes;
+}
+const peak = Math.max.apply(null, props.activity.map(d => d.total));
```
Comment (Johann150) [inline packages/client/src/widgets/activity.calendar.vue]:
```suggestion
const peak = Math.max(...props.activity.map(d => d.total));
```

File: packages/client/src/widgets/activity.chart.vue [hunk 2]
```diff
@@ -40,60 +50,37 @@ function dragClear(fn) {
...
+		let moveTop = me.clientY - clickY;
 
-			// 動かした時
-			dragListen(me => {
-				let moveLeft = me.clientX - clickX;
-				let moveTop = me.clientY - clickY;
+		zoom = baseZoom + (-moveTop / 20);
+		pos = basePos + moveLeft;
+		if (zoom < 1) zoom = 1;
+		if (pos > 0) pos = 0;
```
Comment (Johann150) [inline packages/client/src/widgets/activity.chart.vue]:
```suggestion
		zoom = Math.max(1, baseZoom + (-moveTop / 20));
		pos = Math.min(0, basePos + moveLeft);
```

File: packages/client/src/widgets/activity.chart.vue [hunk 3]
```diff
@@ -40,60 +50,37 @@ function dragClear(fn) {
...
-				if (this.pos < -(((this.data.length - 1) * this.zoom) - this.viewBoxX)) this.pos = -(((this.data.length - 1) * this.zoom) - this.viewBoxX);
+		render();
+	});
+}
 
-				this.render();
-			});
-		}
+function render() {
+	const peak = Math.max.apply(null, props.activity.map(d => d.total));
```
Comment (Johann150) [inline packages/client/src/widgets/activity.chart.vue]:
```suggestion
	const peak = Math.max(...props.activity.map(d => d.total));
```

File: packages/client/src/widgets/server-metric/cpu-mem.vue [hunk 4]
```diff
@@ -69,79 +69,72 @@
...
+	meta: any
+}>();
+
+let viewBoxX: number = $ref(50);
+let viewBoxY: number = $ref(30);
+let stats: any[] = $ref([]);
+let cpuGradientId: string = $ref(uuid());
+let cpuMaskId: string = $ref(uuid());
+let memGradientId: string = $ref(uuid());
+let memMaskId: string = $ref(uuid());
```
Comment (Johann150) [inline packages/client/src/widgets/server-metric/cpu-mem.vue]: These are not written to so you could use `const`s for them.
```suggestion
const cpuGradientId = uuid();
const cpuMaskId = uuid();
const memGradientId = uuid();
const memMaskId = uuid();
```

File: packages/client/src/widgets/server-metric/mem.vue [hunk 5]
```diff
@@ -10,46 +10,34 @@
...
+
+let usage: number = $ref(0);
+let total: number = $ref(0);
+let used: number = $ref(0);
+let free: number = $ref(0);
+
+function onStats(stats) {
+	usage = stats.mem.active / props.meta.mem.total;
+	total = props.meta.mem.total;
+  used = stats.mem.active;
```
Comment (Johann150) [inline packages/client/src/widgets/server-metric/mem.vue]: different indentation from the lines above and below
```suggestion
	used = stats.mem.active;
```

File: packages/client/src/widgets/server-metric/mem.vue [hunk 6]
```diff
@@ -10,46 +10,34 @@
...
+	free = props.meta.mem.total - stats.mem.active;
```
Comment (Johann150) [inline packages/client/src/widgets/server-metric/mem.vue]: I don't know if there might be race conditions of numbers being updated here too, but this at least is shorter.
```suggestion
	free = total - used;
```

**Review by Johann150 — APPROVED** [review]

**Comment by syuilo** [comment]
🙏

**Pull request merged**